llama_stack/providers/inline/agents/meta_reference/openai_responses.py

```diff
@@ -163,7 +256,26 @@ async def create_openai_response(
                     chat_response_content.append(chunk_choice.delta.content or "")
                     if chunk_choice.finish_reason:
                         chunk_finish_reason = chunk_choice.finish_reason
-            assistant_message = OpenAIAssistantMessageParam(content="".join(chat_response_content))
+
+                    if chunk_choice.delta.tool_calls:
+                        for tool_call in chunk_choice.delta.tool_calls:
+                            if tool_call.index not in chat_response_tool_calls:
+                                chat_response_tool_calls[tool_call.index] = OpenAIChatCompletionToolCall(
+                                    **tool_call.model_dump()
+                                )
+                            chat_response_tool_calls[tool_call.index].function.arguments = (
```

Comment:
nit: I think this can be written a bit more compactly perhaps by assigning an intermediate variable

Reply:
Yes, you're right. I added a code comment here about what this does and an intermediate variable as tool call args are accumulated.